app/src/organisms/Devices/ProtocolRun/ProtocolRunSetup.tsx

```diff
@@ -133,8 +136,16 @@ export function ProtocolRunSetup({
   })
 
   if (robot == null) return null
-  const hasLiquids =
-    protocolAnalysis != null && protocolAnalysis.liquids?.length > 0
+
+  const liquids = protocolAnalysis?.liquids ?? []
+
+  const liquidsInLoadOrder =
+    protocolAnalysis != null
+      ? parseLiquidsInLoadOrder(liquids, protocolAnalysis.commands)
+      : []
+
+  const hasLiquids = protocolAnalysis != null && liquidsInLoadOrder.length > 0
```

Comment:
I think hasLiquieds just needs to check liquidsInLoadOrder.length.

```suggestion
  const hasLiquids =  liquidsInLoadOrder.length > 0
```

Reply:
Good catch